// @nx.js/constants/dist/caps.js
export var CapsAlbumStorage;
(function (CapsAlbumStorage) {
    CapsAlbumStorage[CapsAlbumStorage["Nand"] = 0] = "Nand";
    CapsAlbumStorage[CapsAlbumStorage["Sd"] = 1] = "Sd";
})(CapsAlbumStorage || (CapsAlbumStorage = {}));
export var CapsAlbumFileContents;
(function (CapsAlbumFileContents) {
    CapsAlbumFileContents[CapsAlbumFileContents["ScreenShot"] = 0] = "ScreenShot";
    CapsAlbumFileContents[CapsAlbumFileContents["Movie"] = 1] = "Movie";
    CapsAlbumFileContents[CapsAlbumFileContents["ExtraScreenShot"] = 2] = "ExtraScreenShot";
    CapsAlbumFileContents[CapsAlbumFileContents["ExtraMovie"] = 3] = "ExtraMovie";
})(CapsAlbumFileContents || (CapsAlbumFileContents = {}));
export var CapsAlbumFileContentsFlag;
(function (CapsAlbumFileContentsFlag) {
    CapsAlbumFileContentsFlag[CapsAlbumFileContentsFlag["ScreenShot"] = 1] = "ScreenShot";
    CapsAlbumFileContentsFlag[CapsAlbumFileContentsFlag["Movie"] = 2] = "Movie";
})(CapsAlbumFileContentsFlag || (CapsAlbumFileContentsFlag = {}));
