// @nx.js/constants/dist/fs.js
export var FsSaveDataSpaceId;
(function (FsSaveDataSpaceId) {
    FsSaveDataSpaceId[FsSaveDataSpaceId["System"] = 0] = "System";
    FsSaveDataSpaceId[FsSaveDataSpaceId["User"] = 1] = "User";
    FsSaveDataSpaceId[FsSaveDataSpaceId["SdSystem"] = 2] = "SdSystem";
    FsSaveDataSpaceId[FsSaveDataSpaceId["Temporary"] = 3] = "Temporary";
    FsSaveDataSpaceId[FsSaveDataSpaceId["SdUser"] = 4] = "SdUser";
    FsSaveDataSpaceId[FsSaveDataSpaceId["ProperSystem"] = 100] = "ProperSystem";
    FsSaveDataSpaceId[FsSaveDataSpaceId["SafeMode"] = 101] = "SafeMode";
})(FsSaveDataSpaceId || (FsSaveDataSpaceId = {}));
export var FsSaveDataType;
(function (FsSaveDataType) {
    FsSaveDataType[FsSaveDataType["System"] = 0] = "System";
    FsSaveDataType[FsSaveDataType["Account"] = 1] = "Account";
    FsSaveDataType[FsSaveDataType["Bcat"] = 2] = "Bcat";
    FsSaveDataType[FsSaveDataType["Device"] = 3] = "Device";
    FsSaveDataType[FsSaveDataType["Temporary"] = 4] = "Temporary";
    FsSaveDataType[FsSaveDataType["Cache"] = 5] = "Cache";
    FsSaveDataType[FsSaveDataType["SystemBcat"] = 6] = "SystemBcat";
})(FsSaveDataType || (FsSaveDataType = {}));
export var BisPartitionId;
(function (BisPartitionId) {
    BisPartitionId[BisPartitionId["BootPartition1Root"] = 0] = "BootPartition1Root";
    BisPartitionId[BisPartitionId["BootPartition2Root"] = 10] = "BootPartition2Root";
    BisPartitionId[BisPartitionId["UserDataRoot"] = 20] = "UserDataRoot";
    BisPartitionId[BisPartitionId["BootConfigAndPackage2Part1"] = 21] = "BootConfigAndPackage2Part1";
    BisPartitionId[BisPartitionId["BootConfigAndPackage2Part2"] = 22] = "BootConfigAndPackage2Part2";
    BisPartitionId[BisPartitionId["BootConfigAndPackage2Part3"] = 23] = "BootConfigAndPackage2Part3";
    BisPartitionId[BisPartitionId["BootConfigAndPackage2Part4"] = 24] = "BootConfigAndPackage2Part4";
    BisPartitionId[BisPartitionId["BootConfigAndPackage2Part5"] = 25] = "BootConfigAndPackage2Part5";
    BisPartitionId[BisPartitionId["BootConfigAndPackage2Part6"] = 26] = "BootConfigAndPackage2Part6";
    BisPartitionId[BisPartitionId["CalibrationBinary"] = 27] = "CalibrationBinary";
    BisPartitionId[BisPartitionId["CalibrationFile"] = 28] = "CalibrationFile";
    BisPartitionId[BisPartitionId["SafeMode"] = 29] = "SafeMode";
    BisPartitionId[BisPartitionId["User"] = 30] = "User";
    BisPartitionId[BisPartitionId["System"] = 31] = "System";
    BisPartitionId[BisPartitionId["SystemProperEncryption"] = 32] = "SystemProperEncryption";
    BisPartitionId[BisPartitionId["SystemProperPartition"] = 33] = "SystemProperPartition";
    BisPartitionId[BisPartitionId["SignedSystemPartitionOnSafeMode"] = 34] = "SignedSystemPartitionOnSafeMode";
    BisPartitionId[BisPartitionId["DeviceTreeBlob"] = 35] = "DeviceTreeBlob";
    BisPartitionId[BisPartitionId["System0"] = 36] = "System0";
})(BisPartitionId || (BisPartitionId = {}));
