// @nx.js/constants/dist/gamepad.js
/**
 * Mapping of the button indicies on a `Gamepad` to the
 * corresponding name of the button on the controller.
 *
 * @example
 *
 * ```typescript
 * import { Button } from '@nx.js/constants';
 *
 * const pad = navigator.getGamepads()[0];
 * if (pad.buttons[Button.A].pressed) {
 *   console.log('Button A is pressed');
 * }
 * ```
 */
export var Button;
(function (Button) {
    Button[Button["B"] = 0] = "B";
    Button[Button["A"] = 1] = "A";
    Button[Button["Y"] = 2] = "Y";
    Button[Button["X"] = 3] = "X";
    Button[Button["L"] = 4] = "L";
    Button[Button["R"] = 5] = "R";
    Button[Button["ZL"] = 6] = "ZL";
    Button[Button["ZR"] = 7] = "ZR";
    Button[Button["Minus"] = 8] = "Minus";
    Button[Button["Plus"] = 9] = "Plus";
    Button[Button["StickL"] = 10] = "StickL";
    Button[Button["StickR"] = 11] = "StickR";
    Button[Button["Up"] = 12] = "Up";
    Button[Button["Down"] = 13] = "Down";
    Button[Button["Left"] = 14] = "Left";
    Button[Button["Right"] = 15] = "Right";
})(Button || (Button = {}));
