// @nx.js/constants/dist/hid.js
/// HidNpadButton
export var HidNpadButton;
(function (HidNpadButton) {
    HidNpadButton[HidNpadButton["A"] = 1] = "A";
    HidNpadButton[HidNpadButton["B"] = 2] = "B";
    HidNpadButton[HidNpadButton["X"] = 4] = "X";
    HidNpadButton[HidNpadButton["Y"] = 8] = "Y";
    HidNpadButton[HidNpadButton["StickL"] = 16] = "StickL";
    HidNpadButton[HidNpadButton["StickR"] = 32] = "StickR";
    HidNpadButton[HidNpadButton["L"] = 64] = "L";
    HidNpadButton[HidNpadButton["R"] = 128] = "R";
    HidNpadButton[HidNpadButton["ZL"] = 256] = "ZL";
    HidNpadButton[HidNpadButton["ZR"] = 512] = "ZR";
    HidNpadButton[HidNpadButton["Plus"] = 1024] = "Plus";
    HidNpadButton[HidNpadButton["Minus"] = 2048] = "Minus";
    HidNpadButton[HidNpadButton["Left"] = 4096] = "Left";
    HidNpadButton[HidNpadButton["Up"] = 8192] = "Up";
    HidNpadButton[HidNpadButton["Right"] = 16384] = "Right";
    HidNpadButton[HidNpadButton["Down"] = 32768] = "Down";
    HidNpadButton[HidNpadButton["StickLLeft"] = 65536] = "StickLLeft";
    HidNpadButton[HidNpadButton["StickLUp"] = 131072] = "StickLUp";
    HidNpadButton[HidNpadButton["StickLRight"] = 262144] = "StickLRight";
    HidNpadButton[HidNpadButton["StickLDown"] = 524288] = "StickLDown";
    HidNpadButton[HidNpadButton["StickRLeft"] = 1048576] = "StickRLeft";
    HidNpadButton[HidNpadButton["StickRUp"] = 2097152] = "StickRUp";
    HidNpadButton[HidNpadButton["StickRRight"] = 4194304] = "StickRRight";
    HidNpadButton[HidNpadButton["StickRDown"] = 8388608] = "StickRDown";
    HidNpadButton[HidNpadButton["LeftSL"] = 16777216] = "LeftSL";
    HidNpadButton[HidNpadButton["LeftSR"] = 33554432] = "LeftSR";
    HidNpadButton[HidNpadButton["RightSL"] = 67108864] = "RightSL";
    HidNpadButton[HidNpadButton["RightSR"] = 134217728] = "RightSR";
    HidNpadButton[HidNpadButton["Palma"] = 268435456] = "Palma";
    HidNpadButton[HidNpadButton["Verification"] = 536870912] = "Verification";
    HidNpadButton[HidNpadButton["HandheldLeftB"] = 1073741824] = "HandheldLeftB";
    HidNpadButton[HidNpadButton["LagonCLeft"] = -2147483648] = "LagonCLeft";
    HidNpadButton[HidNpadButton["LagonCUp"] = 1] = "LagonCUp";
    HidNpadButton[HidNpadButton["LagonCRight"] = 2] = "LagonCRight";
    HidNpadButton[HidNpadButton["LagonCDown"] = 4] = "LagonCDown";
    HidNpadButton[HidNpadButton["AnyLeft"] = 1118208] = "AnyLeft";
    HidNpadButton[HidNpadButton["AnyUp"] = 2236416] = "AnyUp";
    HidNpadButton[HidNpadButton["AnyRight"] = 4472832] = "AnyRight";
    HidNpadButton[HidNpadButton["AnyDown"] = 8945664] = "AnyDown";
    HidNpadButton[HidNpadButton["AnySL"] = 83886080] = "AnySL";
    HidNpadButton[HidNpadButton["AnySR"] = 167772160] = "AnySR";
})(HidNpadButton || (HidNpadButton = {}));
/// DeviceType (system)
export var HidDeviceTypeBits;
(function (HidDeviceTypeBits) {
    HidDeviceTypeBits[HidDeviceTypeBits["FullKey"] = 1] = "FullKey";
    HidDeviceTypeBits[HidDeviceTypeBits["DebugPad"] = 2] = "DebugPad";
    HidDeviceTypeBits[HidDeviceTypeBits["HandheldLeft"] = 4] = "HandheldLeft";
    HidDeviceTypeBits[HidDeviceTypeBits["HandheldRight"] = 8] = "HandheldRight";
    HidDeviceTypeBits[HidDeviceTypeBits["JoyLeft"] = 16] = "JoyLeft";
    HidDeviceTypeBits[HidDeviceTypeBits["JoyRight"] = 32] = "JoyRight";
    HidDeviceTypeBits[HidDeviceTypeBits["Palma"] = 64] = "Palma";
    HidDeviceTypeBits[HidDeviceTypeBits["LarkHvcLeft"] = 128] = "LarkHvcLeft";
    HidDeviceTypeBits[HidDeviceTypeBits["LarkHvcRight"] = 256] = "LarkHvcRight";
    HidDeviceTypeBits[HidDeviceTypeBits["LarkNesLeft"] = 512] = "LarkNesLeft";
    HidDeviceTypeBits[HidDeviceTypeBits["LarkNesRight"] = 1024] = "LarkNesRight";
    HidDeviceTypeBits[HidDeviceTypeBits["HandheldLarkHvcLeft"] = 2048] = "HandheldLarkHvcLeft";
    HidDeviceTypeBits[HidDeviceTypeBits["HandheldLarkHvcRight"] = 4096] = "HandheldLarkHvcRight";
    HidDeviceTypeBits[HidDeviceTypeBits["HandheldLarkNesLeft"] = 8192] = "HandheldLarkNesLeft";
    HidDeviceTypeBits[HidDeviceTypeBits["HandheldLarkNesRight"] = 16384] = "HandheldLarkNesRight";
    HidDeviceTypeBits[HidDeviceTypeBits["Lucia"] = 32768] = "Lucia";
    HidDeviceTypeBits[HidDeviceTypeBits["Lagon"] = 65536] = "Lagon";
    HidDeviceTypeBits[HidDeviceTypeBits["Lager"] = 131072] = "Lager";
    HidDeviceTypeBits[HidDeviceTypeBits["System"] = -2147483648] = "System";
})(HidDeviceTypeBits || (HidDeviceTypeBits = {}));
/// HID controller styles
export var HidNpadStyleTag;
(function (HidNpadStyleTag) {
    HidNpadStyleTag[HidNpadStyleTag["FullKey"] = 1] = "FullKey";
    HidNpadStyleTag[HidNpadStyleTag["Handheld"] = 2] = "Handheld";
    HidNpadStyleTag[HidNpadStyleTag["JoyDual"] = 4] = "JoyDual";
    HidNpadStyleTag[HidNpadStyleTag["JoyLeft"] = 8] = "JoyLeft";
    HidNpadStyleTag[HidNpadStyleTag["JoyRight"] = 16] = "JoyRight";
    HidNpadStyleTag[HidNpadStyleTag["Gc"] = 32] = "Gc";
    HidNpadStyleTag[HidNpadStyleTag["Palma"] = 64] = "Palma";
    HidNpadStyleTag[HidNpadStyleTag["Lark"] = 128] = "Lark";
    HidNpadStyleTag[HidNpadStyleTag["HandheldLark"] = 256] = "HandheldLark";
    HidNpadStyleTag[HidNpadStyleTag["Lucia"] = 512] = "Lucia";
    HidNpadStyleTag[HidNpadStyleTag["Lagon"] = 1024] = "Lagon";
    HidNpadStyleTag[HidNpadStyleTag["Lager"] = 2048] = "Lager";
    HidNpadStyleTag[HidNpadStyleTag["SystemExt"] = 536870912] = "SystemExt";
    HidNpadStyleTag[HidNpadStyleTag["System"] = 1073741824] = "System";
})(HidNpadStyleTag || (HidNpadStyleTag = {}));
